packages/bundler/src/modules/Types.ts

```diff
@@ -42,6 +42,11 @@ export interface SlotMap {
  * map of storage
  * for each address, either a root hash, or a map of slot:value
  */
+export interface KnownAccount {
+  StorageRoot?: string;
+  StorageSlots?: SlotMap;
+}
+
 export interface StorageMap {
-  [address: string]: string | SlotMap
+  [address: string]: KnownAccount;
```

Comment:
this is a deviation from the spec of the sendRawTransactionConditional, which is already implemented by polygon and arbitrum. do you think we should change it now?
Note that storageSlots and storageRoot are mutually exclusive, which is was it was added as a "union" in the first place.

Reply:
Ah ha, I implemented it based on [Optimism](https://hackmd.io/Jx1B9GLiQ-eltQe7L0FwzQ) , [code](https://github.com/ethereum/go-ethereum/compare/master...tynes:go-ethereum:eip4337#diff-aa2f8b6f5f1dbb32141ae75d1651e332ceec674cba802eec206757c2b93dc382R114) . I just looked at the implementation on Arb, and indeed, there are some differences. I will discuss with them later to see where the problem lies.